Add onnx triton params

glebashnik opened the pull request:
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Implements batching, concurrency and config override params to embedders using onnx.
Some refactoring to DRY the code.
Still contains repetitive code for handling common parameters in embedder configs.
Improved formatting.
Commits will be squashed.

glebashnik committed:
wip, adding triton onnx params

glebashnik committed:
wip

glebashnik committed:
wip onnx triton params

glebashnik committed:
Fixed mode integration embedder

glebashnik committed:
Minor refactoring of handling onnxoptions.

glebashnik committed:
Fixed splade embedder config params

glebashnik committed:
Improved formatting.
Minor fixies.

glebashnik committed:
Added config model unit tests for onnx and triton params.

glebashnik committed:
Added unit tests for TritonOnnxRuntime evaluator configuration.

glebashnik committed:
Adds unittests for building evaluator options from embedder configs.

bjorncs commented on config-model-api/src/main/java/com/yahoo/config/model/api/OnnxModelOptions.java:
```diff
@@ -1,6 +1,10 @@
 
+import com.yahoo.config.FileReference;
+
+import javax.swing.text.html.Option;
```
Dangling import

glebashnik replied:
Removed

bjorncs commented on config-model-api/src/main/java/com/yahoo/config/model/api/OnnxModelOptions.java:
```diff
@@ -10,30 +14,92 @@
+        );
     }
 
     public OnnxModelOptions withExecutionMode(String executionMode) {
```
The builder pattern would be more beneficial with the increasing number of parameters

glebashnik replied:
Agree but we need to keep backwards compatibility so not worth it introducing builder here.

glebashnik committed:
Changed delay to use Duration class, also mills in config to long

glebashnik committed:
Removed unused reference

glebashnik committed:
Moves common onnx parameters from embedder configs to a separate config. Implements builder for OnnxModelOptions. Some minor refactoring.

glebashnik commented on configdefinitions/src/vespa/bert-base-embedder.def:
Moved common params to OnnxEvaluatorConfig now.

bjorncs commented on configdefinitions/src/vespa/onnx-evaluator.def:
```diff
@@ -0,0 +1,28 @@
+# Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+namespace=onnx
```
```suggestion
package=ai.vespa.modelintegration.evaluator.config
```

You also need to export the Java package from `configdefinitions` using `package-info.java`

glebashnik committed:
Moved OnnxEvaluatorConfig to another package.
Fixed handling of defaults in OnnxEvaluatorOptions.
Made batching maxDelay optional.
Addressed comments in PR.

glebashnik commented on configdefinitions/src/vespa/bert-base-embedder.def:
```diff
@@ -24,10 +24,3 @@ transformerEndSequenceToken   int default=102
 
 # Output name
 transformerOutput string default=output_0
-
```
Done, added todos in all embedders.

glebashnik committed:
Added back old OnnxModelOptions constructor for backwards compatibility.

glebashnik commented on config-model-api/abi-spec.json:
```diff
@@ -1610,19 +1610,29 @@
     ],
     "methods" : [
       "public void <init>(java.lang.String, int, int, com.yahoo.config.model.api.OnnxModelOptions$GpuDevice)",
-      "public void <init>(java.util.Optional, java.util.Optional, java.util.Optional, java.util.Optional)",
```
Added this constructor back.

glebashnik committed:
Merge branch 'master' into glebashnik/add-onnx-triton-params

glebashnik committed:
Fixed concurrency attribute names in OnnxModelOptions.
Improved formatting.
